hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -391,6 +395,18 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	reportSilecedAlerts(hcluster)
 	reportProxyConfig(hcluster)
 
+	// Add a grace-period finalizer
+	if hcluster.Annotations == nil {
+		hcluster.Annotations = make(map[string]string)
+	}
+
+	if gracePeriod, ok := hcluster.Annotations[hyperv1.HCDestroyGracePeriodAnnotation]; ok && len(gracePeriod) > 0 && hcDestroyGracePeriod == 0 {
```

Comment:
why the == 0 check?

Reply:
I don't follow, when the code interpreter gets here hcDestroyGracePeriod will always have its zero value as it has been declared in line 323 but never initialized, so as far as I can see this check should be removed.